src/main/java/core/basesyntax/dao/CarJdbcDaoImpl.java

```diff
@@ -0,0 +1,187 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement gettingCar = connection.prepareStatement(query)) {
+            gettingCar.setLong(1, id);
+            ResultSet result = gettingCar.executeQuery();
+            if (result.next()) {
+                car = getCar(result, connection);
+            }
+            return Optional.ofNullable(car);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t add car by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT * FROM cars c "
+                + "JOIN manufacturers m on m.manufacturer_id = c.manufacturer_id "
+                + "WHERE c.deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement gettingAll = connection.prepareStatement(query)) {
+            ResultSet result = gettingAll.executeQuery();
+            List<Car> cars = new ArrayList<>();
+            while (result.next()) {
+                cars.add(getCar(result, connection));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get all cars", e);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateQuery = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE car_id = ? AND deleted = FALSE";
+        String deleteQuery = "DELETE FROM cars_drivers WHERE car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updatedCar = connection.prepareStatement(updateQuery);
+                PreparedStatement deletedRelations = connection.prepareStatement(deleteQuery)) {
+            updatedCar.setString(1, car.getModel());
+            updatedCar.setLong(2, car.getManufacturer().getId());
+            updatedCar.setLong(3, car.getId());
+            updatedCar.executeUpdate();
+            deletedRelations.setLong(1, car.getId());
+            deletedRelations.executeUpdate();
+            insertNewRelations(car, connection);
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t update car" + car, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET deleted = TRUE WHERE car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deletedCar = connection.prepareStatement(query)) {
+            deletedCar.setLong(1, id);
+            return deletedCar.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t delete car by id " + id, e);
+        }
+    }
+
+    private Car getCar(ResultSet result, Connection connection) throws SQLException {
+        Long id = result.getObject(1, Long.class);
```

Comment:
every your SELECT query is like `SELECT * ..` which means that you select every column from the query. if that so, try to give unique names to your columns. this will give you option to `getObject` by column `label` and not depend on their order in query